rfcs/rfc-legal-registry-ledger.md

```diff
@@ -0,0 +1,81 @@
+- Start Date: 2019-04-26
+- Target Major Version: File an Annual-Report
+- Author: thor@wolpert.ca
+- Reference Issues: n/a
+- Entity Issue:
+- Implementation PR:
+
+# Summary
+
+The most basic Legal Entity is a business. All forms of businesses are modified through a series of filings. Historically these filings are paper-based and added in succession to a folder forming the history of a business.
+
+The new model will continue with an immutable ledger that operates in a similar model to how banks, and online banking is viewed. This maintains the historical, legislative and most common mental approach for users of this system.
+
+# Basic example
+
+## Transactions and legal entity state
+A transaction class that is treated as a set of immutable entries would provide a structure like the one below:
+
+| id | transaction id | filing type | filing data |
+|----|----------------|-------------|-------------|
+| 10 |  10            |annual report| {filingType:'AR',filing:{date: '2018-04-10'}}|
+| 11 |  11            |annual report correction| {filingType:'AR',filing:{date: '2018-03-29'}}|
+
+Original filings, corrections, amendments, deletions and the state of the legal entity would all be treated as immutable. Changing anything would require new adding a new filing to alter the business entity into the desired state. The domain model then simplifies to only holding the current state of the legal entity.
+
+## Historical state
+
+Historical changes to a legal entity can be seen by retrieving the *filing* that was saved in the filing_data of the transaction.
+
+For ease of determining the historical view of the legal entity at any point in time, a set of shadow domain objects will hold the changes that were applied to the model over time, allowing the domain objects to be rolled backward and forwards through time.
+
+
+# Motivation
+
+This proposal attempts to provide:
+- an approach to maintaining a simple domain model to describe a *legal entity*
+- a filing history that is immutable and easy to audit.
+- a easy way of viewing the state of the entity at any point in time.
+
+Currently the set of systems used by the registry are fairly complex models used to recreate filings by interpolating domain objects and handle the temporal record in model that does not fully take advantage of the technologies used to maintain it.
+
+The goal here will be to provide a system that, in the most common case, returns the current state of entity efficiently, can return the state of the entity at any given point in time, and provide an immutable, audit-able trail of legal filings that were used to alter the state of the legal entity.
+
+
+# Detailed design
+
+The following Domain Model is what will be used to manage the legal state of a Legal Entity.
+
+![Legal Domain Model](rfc-legal-registry-ledger/legal.png)
```

Comment:
II think we need to add Offices on the top line rather than Addresses.  Drop Address down and associate Office and Parties to Addresses.  Later we may want to add Names, Charter as entities.  Charter can be things like Memorandum of Association, Rules for Co-ops.  Alter Memorandum and Alter Rules are existing co-op filings.